cmd/snap-bootstrap/README.md

```diff
@@ -12,10 +14,64 @@ _snap-bootstrap_ is the main executable that is run during the initramfs stage o
 1. _snap-bootstrap_ on kernel and base snap upgrades will also handle updating bootloader environment variables to implement A/B or try-boot functionality.
 1. _snap-bootstrap_ then finally may do some additional setup of the root filesystem such as copying some default files for ephemeral system modes such as recover. 
 
-## Degraded recover mode
+## In depth walkthrough
+
+_snap-bootstrap_ operates differently depending on snapd_recovery_mode, so each mode is considered separately below.
+
+Note however that snap-bootstrap, while containing the largest chunk of the logic for the initramfs, is not alone; there is other machinery at work in the rest of the initramfs that takes over after snap-bootstrap has exited successfully. Thus, there are more steps to fully complete the initramfs operations beyond these steps, and ideally at some point the rest of those steps will be moved into snap-bootstrap so they can be more fully tested and documented, but for now, just see the unit files in the initrd for "the-modeenv" and "the-tool" to follow what happens after snap-bootstrap is done.
+
+Additionally, note that in all modes where encryption is enabled, we may open/unlock the TPM keys (or other security hardware in the case of custom ARM fde-hooks for example) in order to generally do encryption related things and regardless of result, we must _always_ re-seal and re-lock the TPM (or other security hardware) before exiting snap-bootstrap to protect the sealed-keys. This is implemented specifically with `secboot.LockSealedKeys`.
+
+### Install mode
+
+Install mode has the following steps:
+
+1. The first step of the initramfs-mounts command is always to measure the "epoch" of the secboot version that snap-bootstrap is compiled with to the TPM (if one exists). This is for maximum security and to prevent a newer epoch of secboot from being vulnerable to prior versions.
+1. The next step is to pick the first partition to mount as securely as possible, in the case of EFI systems, we will query an EFI variable to indicate the Partition UUID of the disk which the kernel was booted off of, and we use that Partition UUID to identify the partition which should be mounted as ubuntu-seed (since on grub amd64 systems, the kernel is initially booted by mounting the squashfs in grub and then booting the kernel.efi inside the mounted squashfs). If there is no such EFI variable, we fall back to just using the label instead to choose which partition to mount. Although we do have snap-bootstrap ordered to run after udev has fully settled via `After=systemd-udev-settle.service` in the unit file, sometimes we still don't have that Partition UUID device node available in /dev/ by the time we are executing, so we wait in a loop for the device node to appear before giving up.
```

Comment:
```suggestion
1. The next step is to pick the first partition to mount as securely as possible. With EFI systems, we query an EFI variable used to indicate the Partition UUID of the disk which the kernel was booted off. We then use that Partition UUID to identify the partition which should be mounted as ubuntu-seed (since on grub amd64 systems, the kernel is initially booted by mounting the squashfs in grub and then booting the kernel.efi inside the mounted squashfs). If there is no such EFI variable, we fall back to just using the label to choose which partition to mount. Although we do have snap-bootstrap ordered to run after udev has fully settled via `After=systemd-udev-settle.service` in the unit file, sometimes we still don't have that Partition UUID device node available in /dev/ by the time we are executing, so we wait in a loop for the device node to appear before giving up.
```